docs/_ext/validating_code_block.py

```diff
@@ -0,0 +1,46 @@
+from typing import List
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst import directives
+from sphinx.application import Sphinx
+from sphinx.util.docutils import SphinxDirective
+from sphinx.directives.code import CodeBlock
+from sphinx.errors import ExtensionError
+
+import subprocess
+
+class ValidatingCodeBlock(CodeBlock):
+
+    has_content = True
+    required_arguments = CodeBlock.required_arguments
+    optional_arguments = CodeBlock.optional_arguments
+    final_argument_whitespace = CodeBlock.final_argument_whitespace
+    option_spec = {
+        'type-name': directives.unchanged,
+    }
+    option_spec.update(CodeBlock.option_spec)
+
+    def run(self):
+        source, line = self.state_machine.get_source_and_line(self.lineno)
+        # built-in directives.unchanged_required option validator produces a confusing error message
+        if (self.options.get('type-name') == None):
+            raise ExtensionError("Expected type name in: {0} line: {1}".format(source, line))
+
+        process = subprocess.Popen(['bazel', 'run', '//tools/config_validation:validate_fragment', '--', self.options.get('type-name'), "-s", "\n".join(self.content)],
```

Comment:
You might want to try and pass in `BAZEL_BUILD_OPTIONS` from the env to provide some override, but that's less important here than other places.